CodeGeneration/Sources/SyntaxSupport/Node.swift

```diff
@@ -117,8 +117,13 @@ public class Node {
             isOptional: true,
             collectionElementName: unexpectedName
           ),
-          child,
-        ]
+          child
+        ] + (i == children.count - 1 ? [Child(
+          name: "UnexpectedAfter\(child.name)",
+          kind: "UnexpectedNodes",
+          isOptional: true,
+          collectionElementName: "UnexpectedAfter\(child.name)"
+        )] : [])
```

Comment:
I would prefer to add this after the `flatMap` is completed, i.e. have something like `children.enumerated().flatMap { … } + [<unexpectedAfter>]`. I think that reads better.